wicketd/src/http_entrypoints.rs

```diff
@@ -165,13 +166,65 @@ async fn post_start_update(
         )
     })?;
 
+    // Can we update the target SP? We refuse to update if:
+    //
+    // 1. We haven't pulled its state in our inventory (most likely cause: the
+    //    cubby is empty; less likely cause: the SP is misbehaving, which will
+    //    make updating it very unlikely to work anyway)
+    // 2. We have pulled its state but our hardware manager says we can't update
+    //    it (most likely cause: the target is the sled we're currently running
+    //    on; less likely cause: our hardware manager failed to get our local
+    //    identifying information, and it refuses to update this target out of
+    //    an abundance of caution).
+    //
+    // First, get our most-recently-cached inventory view.
+    let inventory = match rqctx.mgs_handle.get_inventory(Vec::new()).await {
+        Ok(inventory) => inventory,
+        Err(GetInventoryError::ShutdownInProgress) => {
+            return Err(HttpError::for_unavail(
+                None,
+                "Server is shutting down".into(),
+            ));
+        }
+        // We didn't specify any SP ids to refresh, so this error is impossible.
+        Err(GetInventoryError::InvalidSpIdentifier) => unreachable!(),
+    };
+
+    // Next, do we have the state of the target SP?
+    let sp_state = match inventory {
+        GetInventoryResponse::Response { inventory, .. } => inventory
+            .sps
+            .into_iter()
+            .filter_map(|sp| if sp.id == target { sp.state } else { None })
+            .next(),
+        GetInventoryResponse::Unavailable => None,
+    };
+
+    match sp_state {
+        Some(sp_state) => {
+            // If we have the state of the SP, are we allowed to update it?
+            if !rqctx.can_update_target_sp(target, &sp_state) {
+                return Err(HttpError::for_bad_request(
+                    None,
+                    "sleds cannot update themselves".into(),
```

Comment:
Nit: This error could be a little confusing in the case of 

```rust
        if self.baseboard == Baseboard::unknown() {
            !is_scrimlet(id)
        } ...
```

We don't *know* we're on the scrimlet, we just wouldn't want to risk it here, right?

Reply:
That's right; from the other comments we may ditch that case entirely, but if we decide to keep it, I could have `can_update_target_sp` return an enum instead to cover all three cases ("definitely yes", "definitely no", "no because we had problems and don't want to risk it") and give different error responses for the latter two.

Reply:
As part of the rework in [bc7aed0](https://github.com/oxidecomputer/omicron/pull/2940/commits/bc7aed05b88f5cba709d97200da0a3d93995b97e) we now return different error messages for these two cases.